main.js

```diff
@@ -31,47 +36,70 @@ if (isMainThread) {
  * 	output - string, path to output file to write the result
  * }
  */
-async function main({ input, output }) {
-	if (! fs.existsSync(input)) {
+async function main({ urls, output }) {
+
+	if (! fs.existsSync(urls)) {
 		console.log('ERROR: no input file found')
 		return
 	}
 	let workersRunning = 0
 	// if file does not exists or empty
-	let isOutFileEmpty = !(fs.existsSync(output) && fs.readFileSync(output, 'utf8').trim())
-	const inputContent = fs.readFileSync(input, 'utf8').split('\n').filter(p => p)
-	for (const path of inputContent) {
-		if (workersRunning > 10) {
-			// just a delay, so we don't have to many workers runnnig
-			await new Promise(resolve => setTimeout(() => resolve(), 1000))
-		}
-		workersRunning++
-		console.log(workersRunning)
-		const worker = new Worker(WORKER_SCRIPT, {
-			workerData: { path }
-		})
-		// Success
-		// @data is JSON objects separated by comma(,)
-		worker.on('message', (data) => {
-			if (data) {
-				if (! isOutFileEmpty) { data = "," + data }
-				isOutFileEmpty = false
-				saveResultsToFile(data, output, () => console.log(`Finished: ${path}`))
-			}
-		})
-		// Error
-		worker.on('error', (err) => {
-			console.log('ERROR! ' + err)
+	IS_OUT_FILE_EMPTY = !(fs.existsSync(output) && fs.readFileSync(output, 'utf8').trim())
+	const urlList = fs.readFileSync(urls, 'utf8').split('\n').filter(p => p)
+	const minDelay = 3 // in seconds
+	const maxDelay = 5 // in seconds
```

Comment:
Can we use a single object for the delay?
```suggestion
	const delay = {
	        min: 3,
	        max: 5
	} // in seconds
```